src/model/search.js

```diff
@@ -83,7 +86,7 @@ export function getTraceSummary(trace: Trace): TraceSummary {
  * @return {TraceSummaries} The `{ traces, maxDuration }` value.
  */
 export function getTraceSummaries(_traces: Trace[]): TraceSummaries {
-  const traces = _traces.map(getTraceSummary);
+  const traces = _traces.map(getTraceSummary).filter(Boolean);
```

Comment:
not sure I am following the typing here: `_traces` is an array of `Trace` objects, how could `getTraceSummary` function be called with an `Error` object?

Reply:
It's an inefficiency or irregularity in the state of static typing in the repo. Some files have flow-types and some do not. 

The type here is really `(Trace | Error)[]`.

Reply:
so maybe the fix should be (using Scala-ish notation)
```
_traces.filterNot(_ instanceof Error).map(getTraceSummary);
```
?

or even do the filtering somewhere upstream, instead of propagating the mixed type deeper.

Reply:
Agreed. At the least, if it's possible for there to be `Error` objects present, the parameter type should be annotated like you described, `getTraceSummaries(_traces: (Trace | Error)[]): TraceSummaries`.

Reply:
@yurishkuro Great point. I will filter then get the summary. I don't want to filter it further upstream because the presence of the error will be useful in resolving #51.

@simonrobb 👍 

Note, the following code was not possible due to Flow flagging it as an error:

```js
const traces = _traces.filter(item => !(item instanceof Error)).map(getTraceSummary);
```